kobo/apps/audit_log/tests/api/v2/test_api_audit_log.py

```diff
@@ -106,11 +156,132 @@ def test_filter_list(self):
         audit_logs_count = AuditLog.objects.count()
         assert response.status_code == status.HTTP_200_OK
         assert audit_logs_count == 2
         assert response.data['count'] == 1
         assert response.data['results'] == expected
+
+
+class ApiAccessLogTestCase(BaseAuditLogTestCase):
+
+    def get_endpoint_basename(self):
+        return 'access-log-list'
+
+    def setUp(self):
+        super().setUp()
+        user1 = User.objects.get(username='someuser')
+        user2 = User.objects.get(username='anotheruser')
+        # generate 3 access logs, 2 for user1, 1 for user2
+        self.create_access_log(user1)
+        self.create_access_log(user1)
+        self.create_access_log(user2)
+
+        # create a random non-auth audit log
+        log = AuditLog.objects.create(
+            user=user1,
+            app_label='foo',
+            model_name='bar',
+            object_id=1,
+            action=AuditAction.DELETE,
+        )
+        log.save()
+        self.assertEqual(AuditLog.objects.count(), 4)
+
+    def test_list_as_anonymous(self):
+        self.client.logout()
+        response = self.client.get(self.url)
+        assert response.status_code == status.HTTP_401_UNAUTHORIZED
+
+    def test_show_user_access_logs(self):
+        user1 = User.objects.get(username='someuser')
+        self.force_login_user(user1)
+        response = self.client.get(self.url)
+        # only return user1's access logs
+        self.assert_audit_log_results_equal(
+            response=response,
+            expected_kwargs={'action': AuditAction.AUTH, 'user': user1},
+        )
+
+    def test_endpoint_ignores_querystring(self):
+        # make sure a user can't get someone else's access logs
+        user1 = User.objects.get(username='someuser')
+        self.force_login_user(user1)
+        response = self.client.get(f'{self.url}?q=user__username:anotheruser')
+        # check we still only got logs for the logged-in user
+        for audit_log_dict in response.data['results']:
+            self.assertEquals(audit_log_dict['username'], 'someuser')
+
+
+class AllApiAccessLogsTestCase(BaseAuditLogTestCase):
+
+    def get_endpoint_basename(self):
+        return 'all-access-logs-list'
+
+    def setUp(self):
+        super().setUp()
+        super_user = User.objects.get(username='admin')
+        user2 = User.objects.get(username='anotheruser')
+        # generate 3 access logs, 2 for superuser, 1 for user2
+        self.create_access_log(super_user)
+        self.create_access_log(super_user)
+        self.create_access_log(user2)
+
+        # create a random non-auth audit log
+        log = AuditLog.objects.create(
+            user=User.objects.get(username='someuser'),
+            app_label='foo',
+            model_name='bar',
+            object_id=1,
+            action=AuditAction.DELETE,
+        )
+        log.save()
+        self.assertEqual(AuditLog.objects.count(), 4)
+
+    def test_list_as_anonymous(self):
```

Comment:
Not specific to this test only, what do you think about having more explicit names?
For example for this test,  something like `test_list_not_authorized_as_anonymous` . IMO, It makes the behaviour more obvious to the reader.